### PR TITLE
7 Allow to skip output from a code cell as long as the tag is used in…

### PR DESCRIPTION
**Pull request by Adamage**
… a comment

### CODE REVIEW

File: tutorials/sst/src/constants.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 TYPE2EXPORTER = {
-    'jupyter': NotebookExporter,
-    'markdown': MarkdownExporter,
-    'purepython': PythonExporter
-}
\ No newline at end of file
+    OutputTypes.JUPYTER_TYPE: NotebookExporter,
+    OutputTypes.MARKDOWN_TYPE: MarkdownExporter,
+    OutputTypes.PUREPYTHON_TYPE: PythonExporter
+}
```
Comment (plutasnyy) [inline tutorials/sst/src/constants.py:15]: :+1:

File: tutorials/sst/sst.py [hunk 2]
```diff
@@ -33,13 +29,24 @@ def cli(source: Path, output: Path, type: str, execute: bool) -> None:
...
     with open(filename, "w") as fpout:
         fpout.write(output_content)
 
 
+def create_filename(exporter: Exporter, output: Path, source: Path) -> Path:
+    filename = str(output) + exporter.file_extension
+    assert not filename == str(source), f'Your source file and the expected output file name are the same: {source}, ' \
+                                        f'specify different outfile name using --output flag.'
+    return filename
+
```
Comment (plutasnyy) [inline tutorials/sst/sst.py:49]: Aren't there different types here? We are returning a string, and declaring a Path

File: tutorials/sst/tests/test_cli_commands.py [hunk 3]
```diff
@@ -34,6 +34,26 @@ def test_cli_positive(cli_runner_instance):
...
+        ['--source', example_input, "--output", test_output, "--type", "markdown", "--execute"]
+    )
+
+    print(result.stdout)
+    print(result.exception)
+
+    assert result.exit_code == 0
+    assert os.path.exists(test_output_with_extension)
+
+    remove_files_if_present([test_output_with_extension])
```
Comment (plutasnyy) [inline tutorials/sst/tests/test_cli_commands.py:54]: Depending on who does the merge first, tests will need to be updated

File: tutorials/sst/tests/static/code_blocks_with_outputs_to_be_removed.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    print(f'Goodbye {name}!')
+goodbye("sunshine")
+# remove_output
+
+"""
+Example third cell - output hidden
+"""
+hello("sunshine")
+# remove_output
+goodbye("sunshine")
```
Comment (plutasnyy) [inline tutorials/sst/tests/static/code_blocks_with_outputs_to_be_removed.py:23]: Now we have the .py, maybe we can upload the expected .md and do a test to see if the output script works as expected?

**Review by plutasnyy — APPROVED** [review]
I'm wondering if we want to remove the line with the tag from the output in .md?

File: tutorials/sst/tests/static/code_blocks_with_outputs_to_be_removed.py (hunk 4 above)
Comment (Adamage) [reply tutorials/sst/tests/static/code_blocks_with_outputs_to_be_removed.py:23]: I can try

File: tutorials/sst/tests/test_cli_commands.py (hunk 3 above)
Comment (Adamage) [reply tutorials/sst/tests/test_cli_commands.py:54]: sure I can go last

File: tutorials/sst/src/constants.py (hunk 1 above)
Comment (Adamage) [reply tutorials/sst/src/constants.py:15]: yes

**Review by plutasnyy — APPROVED** [review]

**Pull request merged**